lib/ecto/associations.ex

```diff
@@ -99,6 +99,24 @@ defmodule Ecto.Associations do
   end
 
   @doc """
+  Checks if an association is loaded.
+
+  ## Examples
+
+    post = Repo.get(Post, 1)
+    Ecto.Associations.loaded?(post.comments) # false
+    post = post |> Repo.preload(:comments)
+    Ecto.Associations.loaded?(post.comments) # true
+
+  """
+  def loaded?(association) do
+    case association do
+      %Ecto.Associations.NotLoaded{} -> false
```

Comment:
Sounds good :) I wasn't sure if I should extract each module to it's own file in an `/associations` directory to clean this up or not.